File: src/ads_runner_client/lib/domain/entities/subscription.dart
import 'package:equatable/equatable.dart';
import '../../core/enums/subscription_tier.dart';

class Subscription extends Equatable {
  final String id;
  final SubscriptionTier tier;
  final String status;
  final DateTime startDate;
  final DateTime renewalDate;
  final double price;
  final List<String> features;

  const Subscription({
    required this.id,
    required this.tier,
    required this.status,
    required this.startDate,
    required this.renewalDate,
    required this.price,
    required this.features,
  });

  @override
  List<Object> get props => [id, tier, status];
}

File: src/AdsRunner.Domain/Entities/Subscription.cs
using AdsRunner.Domain.Common;
using AdsRunner.Domain.Enums;

namespace AdsRunner.Domain.Entities;

public class Subscription : BaseEntity
{
    public SubscriptionTier Tier { get; set; }
    public string Status { get; set; } = string.Empty;
    public DateTime StartDate { get; set; }
    public DateTime RenewalDate { get; set; }
    public decimal Price { get; set; }

    // Foreign keys
    public Guid UserId { get; set; }

    // Navigation properties
    public User User { get; set; } = null!;
    public ICollection<PaymentRecord> PaymentRecords { get; set; } = new List<PaymentRecord>();
}
